### PR TITLE
fix(web): refresh bugs and image-relative overlays with zoom support for faces, OCR, and face editor

### PR DESCRIPTION
**Pull request by midzelis**
https://github.com/user-attachments/assets/448fa01d-6b7d-4081-80b5-9ab583232fdd

https://github.com/user-attachments/assets/44d80b6d-aa5b-4470-8840-56a3fcc5c9be

Split out from #26636 (adaptive image loading) to keep scope down. Leverages the `ContentMetrics` extraction from #26310.

Since #26636 introduced layered image loading (thumbnail/preview/original as separate `` elements), and `zoom-image` only targets single images by default. This PR changes zoom to target the entire `AdaptiveImage` div, and moves face bounding boxes and OCR overlays to image-relative coordinates so the browser handles zoom transforms automatically.

### Benefits
- Eliminates manual coordinate math for zoom/pan compensation — overlays move naturally with the image
- Simpler, more maintainable overlay positioning code

### New features
- Face editor selection rectangle scales and repositions with zoom — the Fabric.js canvas viewport transform mirrors the image zoom state, so the selection box grows and pans naturally when you zoom in/out with the face editor open
- Wheel-to-zoom works through the face editor overlay — a capturing listener on the parent element forwards wheel events from sibling layers to the zoom handler
- Zoom with OCR boxes visible
- Accurate face hover hit-testing at any zoom level (previously broken when zoomed)

### Bug fixes
- Resizing the window with the face editor open now preserves the relative position and proportional size of the face selection rectangle, instead of resetting it to a fixed offset
- Opening/closing the detail panel repositions and resizes the face editor proportionally, treating the panel edge as the viewer boundary
- Face tagging coordinates use object-space values instead of viewport-transformed bounds, so crop coordinates remain accurate regardless of zoom level
- `disablePointer` option name mismatch meant pointer-blocking in face edit mode never worked

### Changes
**Face bounding boxes:** Replace container-level `mousemove` hit-testing with per-face `mouseenter`/`mouseleave` — the browser handles hit-testing since the divs are children of the zoomed image.

**Zoom:** Pass `zoomTarget` to the zoom library, fix `disablePointer` naming. Forward wheel events from sibling overlays via capturing listener. Consolidate cleanup to AbortController.

**Face editor:** Move from photo-viewer to asset-viewer for shared container dimensions. Apply Fabric.js viewport transform to match image zoom state. Scale face rect proportionally on container resize.

**AdaptiveImage:** Move zoom transform (translate + scale) into AdaptiveImage as a CSS transform on an inner wrapper div, so all child overlays inherit the zoom naturally.

**Asset viewer:** Refactor OCR refresh for stacked assets. Move stacked assets UI into viewer content div. Add containerWidth/containerHeight state.

### New Tests
**Face overlay** (`face-overlay.e2e-spec.ts`):
- Face overlay divs render with correct aria labels
- Face border and name tooltip appear on hover
- Face overlays hidden in face edit mode

**OCR** (`ocr.e2e-spec.ts`):
- OCR boxes appear/toggle on button click
- Different OCR boxes shown for different stacked assets

**Unit tests** (`people-utils.spec.ts`):
- `scaleFaceRectOnResize` covering proportional scaling, offset adjustment, and edge cases

### CODE REVIEW

File: web/src/lib/components/asset-viewer/asset-viewer.svelte [hunk 1]
```diff
@@ -358,15 +359,18 @@
...
-  const refresh = async () => {
-    await refreshStack();
+  const refreshOcr = async () => {
     ocrManager.clear();
-    if (!sharedLink) {
-      if (previewStackedAsset) {
-        await ocrManager.getAssetOcr(previewStackedAsset.id);
-      }
-      await ocrManager.getAssetOcr(asset.id);
+    if (sharedLink) {
```
Comment (alextran1502) [inline web/src/lib/components/asset-viewer/asset-viewer.svelte:364]: Nice refactoring!

**Comment by ghost** [comment]
@midzelis check https://github.com/immich-app/immich/discussions/26908

**Comment by danieldietzler** [comment]
@rthrth-svg do not ping other people in multiple places. Also,

> can fix within one of your prs related to refactoring of ocr boxes and faces

is just bad advice. It's a separate feature so it should be done separately

**Comment by ghost** [comment]
> is just bad advice. It's a separate feature so it should be done separately

i've only done that here because this looks like a massive change and adding that fits within this pr as well

https://github.com/immich-app/immich/pull/26678#issuecomment-4031127500

**Comment by alextran1502** [comment]
@rthrth-svg, do you have a Discord handle so we can communicate with you more easily? I appreciate the enthusiastic in trying to fix things but it seems like your communication style can come off as pushy and it would be great if we can have conversation on Discord before you dispatch a fix or to ask for more inputs

**Comment by ghost** [comment]
> communication style can come off as pushy

yeah i get it, but i am trying to remain as genuine as possible (_so do not take comments personal, they're more related to initial design decisions rather than personal insults or anything lol_) especially after reading all the comments of people who are as frustrated (_and even worse considering they've been using it for longer than me_) as me (_i.e. that issue related to perf issues or https://github.com/immich-app/immich/discussions/12022. which can be easily implemented within the existing "folders" view (gallery-viewer.svelte) after generalizing its usage and splitting into components and letting server collect everything within selected months thru years/years (you're already dispatching clients with megabytes of data so wont change much), so im not sure what's the point of abandoning it_)

p.s immich is currently the only app that supports most of the things (_other apps lack support of some things that are presented in one but not in others_) in a single bundle and there's no other alternatives and competing doesn't seem reasonable, (i'd) rather fix it (_push few important fixes and disappear_) considering its somewhat maintained and re-download each time i need to sort things out or perform OCR/face recognition on a set, immich easily can become better tho, has to be modular however rather than an omelette (_yes i understand that immich isn't meant for "everyone" but monopoly + low quality (current state) is bad_)

as of discord, i sent my mate just to figure out whether joining it is worthy or not, tldr at least we figured out that plugins system in future can solve some of the issues that are currently presented, resolution: https://github.com/immich-app/immich/issues/26794#issuecomment-4023278772 (last part)

p.s bit more of offtopic, mate sent me this few days ago and asked to report but i didn't have a proper context to do so, so https://demo.immich.app/photos/364d44c8-1a9e-405c-9ad5-9b0c70fd205b might be viewed as csam under certain conditions, try going through them all and deleting somewhat related to that images

**Review by michelheusschen — COMMENTED** [review]
I've noticed your PRs often contain bugs that could probably be caught with a few minutes of manual testing or self review. Could you spend a bit more time testing changes before opening a PR?

Here's some issues I found with 5 minutes of testing:
- When adding a face box you can't search people
- Face boxes are saved with a wrong offset
- On touch devices, tapping people in the side panel does not show the face overlay
- After editing faces, hovering over people in the side panel no longer shows face overlay
- While editing faces, hovering over people doesn't show the face overlay

**Comment by midzelis** [comment]
> I've noticed your PRs often contain bugs that could probably be caught with a few minutes of manual testing or self review. Could you spend a bit more time testing changes before opening a PR?
> 
> Here's some issues I found with 5 minutes of testing:
> 
> * When adding a face box you can't search people
> * Face boxes are saved with a wrong offset
> * On touch devices, tapping people in the side panel does not show the face overlay
> * After editing faces, hovering over people in the side panel no longer shows face overlay
> * While editing faces, hovering over people doesn't show the face overlay

You're right - this one was poorly tested. Last push fixes the problems you identified - thanks! 

I also fixed a few more: 
- cancel on confirm tag used to close face editor, now it just closes the confirm and leave face editor open
- click on free space will re-center bounding box on the click/touch 

I also noticed that on small mobile devices (iphone) the detail pane is too large - and the face editor modal is too wide, etc. Those changes were split out from here (so that this PR didn't get too big) #26945

**Review by meesfrensel — COMMENTED** [review]
It's not possible to select OCR text anymore; instead, I just drag the image. Note that this was originally also the reason zoom/pan was disabled when showing OCR overlays.

**Comment by midzelis** [comment]
This has been split into two PRs: #27039 and #27040
